doc/source/tutorial/stats/sampling.rst

```diff
@@ -0,0 +1,202 @@
+`np.random.BitGenerator`, `np.random.RandomState`, or
+`np.random.SeedSequence`.
+
+.. warning:: Use of NumPy < 1.19.0 is discouraged as it doesn't have a fast
+             Cython API for generating uniform random numbers and might be
+             too slow for practical use.
+
+All the generators have a common ``rvs`` method that can be used to draw
+samples from the given distribution.
+
+An example of this interface is shown below:
+
+    >>> from scipy.stats import TransformedDensityRejection
+    >>> from math import exp
+    >>> 
+    >>> class StandardNormal:
+    ...     def pdf(self, x: float) -> float:
+    ...         # note that the normalization constant isn't required
+    ...         return exp(-0.5 * x*x)
+    ...     def dpdf(self, x: float) -> float:
+    ...         return -x * exp(-0.5 * x*x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> 
+    >>> urng = np.random.default_rng(0x6be00a9336fc82258fe73bfa672d3d52)
+    >>> rng = TransformedDensityRejection(dist, seed=urng)
+
+As shown in the example, we first initialize a distribution object that
+contains an implementation of the methods required by the generator. In
+our case, we use the :class:`~TransformedDensityRejection` method which
+requires a PDF and its derivative w.r.t. x (i.e. the variate).
+
+.. note:: Note that the methods of the distribution (i.e. ``pdf``,
+          ``dpdf``, etc) need not be vectorized. They should
+          accept and return floats.
+
+.. note:: One can also pass the SciPy distributions as arguments but it can
+          be slow due to validations and expensive NumPy operations.
+          Moreover, it doesn't always have all the information required
+          by some generators like derivative of PDF for TDR method. Also,
+          most of the distributions in SciPy provide a ``rvs`` method which
+          can be used instead.
+
+In the above example, we have set up an object of the
+:class:`~TransformedDensityRejection` method to sample from a
+standard normal distribution. Now, we can start sampling from our
+distribution by calling the ``rvs`` method:
+
+    >>> rng.rvs()
+    -0.6853213192313732
+    >>> rng.rvs((5, 3))
+    array([[ 0.24129542,  1.28709556,  0.43244247],
+           [ 0.88977668,  1.22981663, -0.09820373],
+           [ 1.38588147, -1.06629264,  0.54475257],
+           [-0.55153436, -0.67281406,  0.01830143],
+           [ 1.37709649, -1.33036774,  0.62131967]])
+
+.. note:: Please note the difference between the `rvs` method of the
+          distributions present in :mod:`scipy.stats` and the one provided
+          by these generators. In general, the aim of the generation
+          methods implemented in :class:`~rv_continuous` is to provide
+          generators that work reasonably well for common use cases, e.g.
+          to generate small samples using different shape parameters or to
+          generate large samples for a fixed shape parameter. Depending on
```

Comment:
I am not sure I understand. Does it mean that only some values can be used to generate large samples?

Reply:
> Does it mean that only some values can be used to generate large samples?

I assume you are talking about the line:

```
          to generate small samples using different shape parameters or to
          generate large samples for a fixed shape parameter.
```

I think it is saying this because when we change the shape parameters, the setup step needs to be run again. Hence, it is more common to draw a small number of samples between every setup step. In contrast, when we fix the shape parameters, the setup is run only once and we can draw a large number of samples in the same time. That is what the "*common* use cases" is supposed to mean, I guess. Do you think making this explicit would make it easier for general users to grasp this?

@chrisb83 Do you agree with the above explanation?

Reply:
I think yes. I am trying to read this with not to much prior (and in this case I was not sure what you meant, now it's clear you are talking about freezing the distribution) and it don't really get this otherwise.

Reply:
ok, probably we need to add a bit more information: in random variate generation, you are interested in the fixed and the varying parameter case. 

- fixed: fix a shape parameter and draw samples => in general, this is the easier case since you can run a slow setup step once and then draw samples again and again
- varying: the shape parameter keeps changing when you draw new samples. an example is Gibbs sampling => the distribution of the next sample depends on the current one. in that case, you cannot use an expensive setup. or consider simulating the initial wealth of 100k agents, each drawn from a distribution with a different mean

so choosing the right method strongly depends on the situation...

Reply:
since this is an important topic, we should explain these concepts better. in the UNU.RAN doc, the authors give an indication of the setup and sampling speed, e.g.,

Automatic ratio of uniforms method: Set-up: slow, Sampling: fast
Simple ratio of uniforms method: Set-up: fast, Sampling: slow

I think we should create a table in the tutorial as follows:

| Method | Required inputs | Setup speed | Sampling Speed |
| SimpleRatioUniforms | pdf, ... | fast | slow |

@tirthasheshpatel I can draft a short introduction to explain these concepts better.

Reply:
> since this is an important topic, we should explain these concepts better.

I agree.

> I think we should create a table in the tutorial as follows:

I think a summary table like the one you have made would be very useful. I have also mentioned this information in the tutorial of methods (see `sampling_tdr`, `sampling_dau`).

> I can draft a short introduction to explain these concepts better.

That would be a lot of help @chrisb83, thanks!

Reply:
I have added a simple table @chrisb83 (see ea5e083219a3223b33b2aca3a1c94cb8c9659622). Let me know what you think about it.

Reply:
Great! I would just suggest to create separate tables for continuous and discrete distributions

Reply:
This is really helpful, thanks!

Reply:
@chrisb83 I have separated discrete and continuous methods now.

Reply:
Feel free to resolve this one. On my side it's covered.